### PR TITLE
AB#89372 -fix for national design guide not get read from repository, recovered …

### PR DESCRIPTION
**Pull request by rafalgajewski**
…missing Section 106 ineligible page

### CODE REVIEW

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106AdditionalAffordableHousingCommandHandler.cs [hunk 1]
```diff
@@ -20,15 +20,11 @@ public Task<OperationResult> Handle(ProvideSection106AdditionalAffordableHousing
...
             {
                 var currentSection106 = site.Section106;
-                var newSection106 = new Section106(
-                                            currentSection106.GeneralAgreement,
-                                            currentSection106.AffordableHousing,
-                                            currentSection106.OnlyAffordableHousing,
-                                            request.AdditionalAffordableHousing,
-                                            currentSection106.CapitalFundingEligibility,
-                                            currentSection106.LocalAuthorityConfirmation);
+                if (request.AdditionalAffordableHousing.HasValue && request.AdditionalAffordableHousing != currentSection106.AdditionalAffordableHousing)
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106AdditionalAffordableHousingCommandHandler.cs:23]: Are those checks needed? ValueObjects has GetAtomicValues method which takes care of such things. `ModificationTracker` will not mark Entity as modified if two different instances of ValueObjects have the same values.

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs [hunk 2]
```diff
@@ -95,6 +95,36 @@ public bool IsIneligible()
         return IsIneligibleDueToAffordableHousing() || IsIneligibleDueToCapitalFundingGuide();
     }
 
+    public Section106 WithGeneralAgreement(bool? generalAgreement)
+    {
+        return new Section106(generalAgreement);
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Site/ValueObjects/Section106.cs:100]: You should set all fields in every method. Check `MilestoneTranches`

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Views/Site/Section106Ineligible.cshtml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+@using HE.Investments.Common.WWW.Components.FormHeader;
+@using HE.Investment.AHP.WWW.Views.Site;
+@model Section106Dto
+@{
+    ViewData["Title"] = SitePageTitles.Section106Ineligible;
+}
+
+<div class="govuk-grid-column-three-quarters">
+    <vc:ahp-workflow-back-button current-page="@SiteWorkflowState.Section106Ineligible" />
+    <vc:form-header title="@SitePageTitles.Section106Ineligible" />
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/Site/Section106Ineligible.cshtml:12]: page-header

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs [hunk 4]
```diff
@@ -130,6 +130,9 @@ private void ConfigureTransitions()
             .PermitIf(Trigger.Back, SiteWorkflowState.Section106OnlyAffordableHousing, () => _siteModel?.Section106?.OnlyAffordableHousing == true)
             .PermitIf(Trigger.Back, SiteWorkflowState.Section106AffordableHousing, () => _siteModel?.Section106?.OnlyAffordableHousing == null);
 
+        _machine.Configure(SiteWorkflowState.Section106Ineligible)
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:133]: missing tests

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs [hunk 5]
```diff
@@ -209,9 +212,11 @@ private void ConfigureTransitions()
     private bool IsTenderForWorksContractOrContractingHasNotYetBegun() => _siteModel?.TenderingStatusDetails.TenderingStatus is SiteTenderingStatus.TenderForWorksContract or
         SiteTenderingStatus.ContractingHasNotYetBegun;
 
-    private bool IsSection106EligibleWithAdditionalAffordableHousing() => _siteModel?.Section106?.CapitalFundingEligibility == false && _siteModel?.Section106?.AdditionalAffordableHousing == true;
+    private bool IsSection106EligibleWithAdditionalAffordableHousing() => _siteModel?.Section106?.AdditionalAffordableHousing == true
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:215]: missing tests

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106AdditionalAffordableHousingCommandHandler.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106AdditionalAffordableHousingCommandHandler.cs:23]: Those are needed to recognize this situation when user went back and changed one of the values - then we need to clear all the following section 106 choices. But I think moving this logic to value object is a good idea. Will do that.

**Pull request closed without merging**